mlflow/server/auth/sqlalchemy_store.py

```diff
@@ -140,3 +145,167 @@ def list_users(self) -> List[User]:
         with self.ManagedSessionMaker() as session:
             users = session.query(SqlUser).all()
             return [u.to_mlflow_entity() for u in users]
+
+    def update_user(self, username: str, password: str = None, is_admin: bool = None) -> User:
+        with self.ManagedSessionMaker() as session:
+            user = self._get_user(session, username)
+            if password is not None:
+                pwhash = generate_password_hash(password)
+                user.password_hash = pwhash
+            if is_admin is not None:
+                user.is_admin = is_admin
+            return user.to_mlflow_entity()
+
+    def delete_user(self, username: str):
+        with self.ManagedSessionMaker() as session:
+            user = self._get_user(session, username)
+            session.delete(user)
+
+    def create_experiment_permission(
+        self, experiment_id: str, username: str, permission: str
+    ) -> ExperimentPermission:
+        _validate_permission(permission)
+        with self.ManagedSessionMaker() as session:
+            try:
+                user = self._get_user(session, username=username)
+                perm = SqlExperimentPermission(
+                    experiment_id=experiment_id, user_id=user.id, permission=permission
+                )
+                session.add(perm)
+                session.flush()
+                return perm.to_mlflow_entity()
+            except IntegrityError as e:
+                raise MlflowException(
+                    f"Experiment permission (experiment_id={experiment_id}, username={username}) "
+                    f"already exists. Error: {e}",
+                    RESOURCE_ALREADY_EXISTS,
+                )
+
+    @classmethod
+    def _get_experiment_permission(
+        cls, session, experiment_id: str, username: str
+    ) -> SqlExperimentPermission:
+        try:
+            user = cls._get_user(session, username=username)
+            return (
+                session.query(SqlExperimentPermission)
+                .filter(SqlExperimentPermission.experiment_id == experiment_id)
+                .filter(SqlExperimentPermission.user_id == user.id)
```

Comment:
```suggestion
                .filter(SqlExperimentPermission.experiment_id == experiment_id, (SqlExperimentPermission.user_id == user.id)
```

Can we combine these two con

Reply:
updated